In [1]:
# LIBRERIAS UTILIZADAS
import numpy as np
import pandas as pd
import gmaps
import gmaps.datasets

In [2]:
listing = pd.read_csv("df_final.csv")

In [3]:
listing["latitude"].describe()

count    13252.000000
mean        41.392081
std          0.014308
min         41.352610
25%         41.380888
50%         41.389660
75%         41.401640
max         41.461930
Name: latitude, dtype: float64

Observamos un maximo de 41.46193 y un minimo de 41.35261, por lo tanto escogeremos un intervalo de [41.30 , 41.50]

In [4]:
listing["longitude"].describe()

count    13252.000000
mean         2.167541
std          0.017475
min          2.091590
25%          2.157437
50%          2.168605
75%          2.177623
max          2.227710
Name: longitude, dtype: float64

Observamos un maximo de 2.22771 y un minimo de 2.09159, por lo tanto escogeremos un intervalo de [2.03, 2.28]

In [5]:
gmaps.configure(api_key="AIzaSyD2q-mG8Phl5JljfPqk0Wtv4fWQrH_dvas")

Empezamos realizando un heatmap para una variable que contenga el número de viviendas que hay en alquiler.

In [6]:
Lon = np.arange(2.03, 2.28, 0.0025) 
Lat = np.arange(41.30, 41.50, 0.00201) 
listing_counts = np.zeros((100,100)) 

In [7]:
for a in range(len(listing)):
  for b1 in range(100):
    if Lat[b1] - 0.00100 <= listing["latitude"].values[a] < Lat[b1] + 0.00100:
      for b2 in range(100):
        if Lon[b2] - 0.00125 <= listing["longitude"].values[a] < Lon[b2] + 0.00125:
          listing_counts[b1,b2] += 1

In [8]:
longitude_values = [Lon,]*100 
latitude_values = np.repeat(Lat,100) 
listing_counts.resize((10000,)) 

In [9]:
heatmap_data = {'Counts': listing_counts, 'latitude': latitude_values, 'longitude' : np.concatenate(longitude_values)} 
df = pd.DataFrame(data=heatmap_data) 

In [10]:
locations = df[['latitude', 'longitude']] 
weights = df['Counts'] 
fig = gmaps.figure() 
heatmap_layer = gmaps.heatmap_layer(locations, weights=weights) 
fig.add_layer(gmaps.heatmap_layer(locations, weights=weights))
fig

Figure(layout=FigureLayout(height='420px'))

Haremos ahora un gráfico para la media de precio.

In [11]:
listing_price = np.zeros((100,100))

In [12]:
for a in range(len(listing)):
  for b1 in range(100):
    if Lat[b1] - 0.00100 <= listing["latitude"].values[a] < Lat[b1] + 0.00100:
      for b2 in range(100):
        if Lon[b2] - 0.00125 <= listing["longitude"].values[a] < Lon[b2] + 0.00125:
          listing_price[b1,b2] += listing["price"].values[a]

In [13]:
longitude_values = [Lon,]*100 
latitude_values = np.repeat(Lat,100) 
listing_price.resize((10000,)) 

In [14]:
listing_mean_price = listing_price/listing_counts
listing_mean_price[np.isnan(listing_mean_price)] = 0

C:\Users\Andreu\AppData\Local\Temp\ipykernel_492\705762398.py:1: RuntimeWarning: invalid value encountered in true_divide
  listing_mean_price = listing_price/listing_counts


In [15]:
heatmap_data = {'Price': listing_mean_price, 'latitude': latitude_values, 'longitude' : np.concatenate(longitude_values)} 
df = pd.DataFrame(data=heatmap_data) 

In [16]:
locations = df[['latitude', 'longitude']] 
weights = df['Price'] 
fig = gmaps.figure() 
heatmap_layer = gmaps.heatmap_layer(locations, weights=weights) 
fig.add_layer(gmaps.heatmap_layer(locations, weights=weights))
fig

Figure(layout=FigureLayout(height='420px'))

Finalmente realizaremos mapas de calor para la media de las dos variables respuesta, "days_booked" y "reviews_diff". Empezamos por "days_booked".

In [17]:
listing_days = np.zeros((100,100))

In [18]:
for a in range(len(listing)):
  for b1 in range(100):
    if Lat[b1] - 0.00100 <= listing["latitude"].values[a] < Lat[b1] + 0.00100:
      for b2 in range(100):
        if Lon[b2] - 0.00125 <= listing["longitude"].values[a] < Lon[b2] + 0.00125:
          listing_days[b1,b2] += listing["days_booked"].values[a]

In [19]:
longitude_values = [Lon,]*100 
latitude_values = np.repeat(Lat,100) 
listing_days.resize((10000,)) 

In [20]:
listing_mean_days = listing_days/listing_counts
listing_mean_days[np.isnan(listing_mean_days)] = 0

C:\Users\Andreu\AppData\Local\Temp\ipykernel_492\1967739511.py:1: RuntimeWarning: invalid value encountered in true_divide
  listing_mean_days = listing_days/listing_counts


In [21]:
heatmap_data = {'days booked': listing_mean_days, 'latitude': latitude_values, 'longitude' : np.concatenate(longitude_values)} 
df = pd.DataFrame(data=heatmap_data) 

In [22]:
locations = df[['latitude', 'longitude']] 
weights = df['days booked'] 
fig = gmaps.figure() 
heatmap_layer = gmaps.heatmap_layer(locations, weights=weights) 
fig.add_layer(gmaps.heatmap_layer(locations, weights=weights))
fig

Figure(layout=FigureLayout(height='420px'))

Realizamos ahora para "reviews_diff".

Realizaremos una transformacion previa en "reviews_diff", ya que el mapa sale mal debido a que hay un par de valores muy grandes en esta, por lo tanto fijaremos el maximo de esta a 20.

In [44]:
listing["reviews_diff"].where(listing["reviews_diff"] <= 25, 25, inplace=True)
listing["reviews_diff"].where(listing["reviews_diff"] >= 0, 0, inplace=True)

In [45]:
listing_reviews = np.zeros((100,100))

In [46]:
for a in range(len(listing)):
  for b1 in range(100):
    if Lat[b1] - 0.00100 <= listing["latitude"].values[a] < Lat[b1] + 0.00100:
      for b2 in range(100):
        if Lon[b2] - 0.00125 <= listing["longitude"].values[a] < Lon[b2] + 0.00125:
          listing_reviews[b1,b2] += listing["reviews_diff"].values[a]

In [47]:
longitude_values = [Lon,]*100 
latitude_values = np.repeat(Lat,100) 
listing_reviews.resize((10000,)) 
listing_reviews[listing_reviews<0] = 0

In [48]:
listing_mean_reviews = listing_reviews/listing_counts
listing_mean_reviews[np.isnan(listing_mean_reviews)] = 0

C:\Users\Andreu\AppData\Local\Temp\ipykernel_492\3335887488.py:1: RuntimeWarning: invalid value encountered in true_divide
  listing_mean_reviews = listing_reviews/listing_counts


In [49]:
heatmap_data = {'reviews diff': listing_mean_reviews, 'latitude': latitude_values, 'longitude' : np.concatenate(longitude_values)} 
df = pd.DataFrame(data=heatmap_data) 

In [50]:
locations = df[['latitude', 'longitude']] 
weights = df['reviews diff'] 
fig = gmaps.figure() 
heatmap_layer = gmaps.heatmap_layer(locations, weights=weights) 
fig.add_layer(gmaps.heatmap_layer(locations, weights=weights))
fig

Figure(layout=FigureLayout(height='420px'))

Para la variable host_acceptance_rate.

In [53]:
listing_acceptance = np.zeros((100,100))

In [54]:
for a in range(len(listing)):
  for b1 in range(100):
    if Lat[b1] - 0.00100 <= listing["latitude"].values[a] < Lat[b1] + 0.00100:
      for b2 in range(100):
        if Lon[b2] - 0.00125 <= listing["longitude"].values[a] < Lon[b2] + 0.00125:
          listing_acceptance[b1,b2] += listing["host_acceptance_rate"].values[a]

In [55]:
longitude_values = [Lon,]*100 
latitude_values = np.repeat(Lat,100) 
listing_acceptance.resize((10000,)) 
listing_acceptance[listing_acceptance<0] = 0

In [56]:
listing_mean_acceptance = listing_acceptance/listing_counts
listing_mean_acceptance[np.isnan(listing_mean_acceptance)] = 0

C:\Users\Andreu\AppData\Local\Temp\ipykernel_9724\3494744952.py:1: RuntimeWarning: invalid value encountered in true_divide
  listing_mean_acceptance = listing_acceptance/listing_counts


In [57]:
heatmap_data = {'host acceptance rate': listing_mean_acceptance, 'latitude': latitude_values, 'longitude' : np.concatenate(longitude_values)} 
df = pd.DataFrame(data=heatmap_data) 

In [58]:
locations = df[['latitude', 'longitude']] 
weights = df['host acceptance rate'] 
fig = gmaps.figure() 
heatmap_layer = gmaps.heatmap_layer(locations, weights=weights) 
fig.add_layer(gmaps.heatmap_layer(locations, weights=weights))
fig

Figure(layout=FigureLayout(height='420px'))